### ISSUE TITLE
When search query contains "%" the UI (facetting) breaks

### ISSUE THREAD

**Issue by legonzian** [SAP/spartacus#14558]
**Describe the bug**
When search string contains "%" the FacetService.getLinkParams throws this error:

`ERROR URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at HttpUrlEncodingCodec.decodeValue (http.js:324)
    at FacetService.getLinkParams (spartacus-storefront.js:35475)
    at ActiveFacetsComponent.getLinkParams (spartacus-storefront.js:35505)
    at ActiveFacetsComponent_ng_container_0_a_2_Template (active-facets.component.html:7)
    at executeTemplate (core.js:7511)
    at refreshView (core.js:7380)
    at refreshEmbeddedViews (core.js:8481)
    at refreshView (core.js:7404)
    at refreshEmbeddedViews (core.js:8481)`

**Tell us the version of Spartacus**
- Library version: 3.3.x

**To Reproduce**
1. Go to any page that has the search-box.component present
2. Enter some string containg percent sign (e.g. "cam -25%")
3. Hit enter
4. UI on Search Page will stop rendering and throws error periodically

**Expected behavior**
Search Page with corresponding results

**Screenshots**
![image](https://user-images.githubusercontent.com/26231533/145193332-6db08259-aca1-484c-a6f5-8bb8e0f7bfc2.png)

**Desktop (please complete the following information):**
 - OS: iOs
 - Browser: chrome
 - Version: 96.0.4664.93

**Additional context**

**Comment by legonzian** [comment]
Fixed it by providing custom FacetService and overriding getLinkParams-Method:

`  getLinkParams(query: string): { [key: string]: string } {
    query = this.decodeUriComponentSafe(query);
    return super.getLinkParams(query);
  }

  decodeUriComponentSafe(query: string): string {
    // Added replace function on query parameter to safely decode URI component, when search query contains "%":
    // https://stackoverflow.com/questions/7449588/why-does-decodeuricomponent-lock-up-my-browser
    if(!query.startsWith(":")) {
      let queryChunks = query.split(":");
      queryChunks[0] = queryChunks[0].replace(/%(?![0-9][0-9a-fA-F]+)/g, "%25");
      query = queryChunks.join(":");
    }
    return query;
  }`

**Comment by Xymmer** [comment]
fix in 5.0, backport 4.x

**Comment by giancorderoortiz** [comment]
Should test for all special characters, and regex. 
(Breaks facets)

**Comment by KateChuen** [comment]
Tested in develop branch locally pointing to dev7 backend (using yarn).
- This specific issue is fixed. Facets are ok now when we search for: 
<img width="1236" alt="Screen Shot 2022-04-01 at 3 43 53 PM" src="https://user-images.githubusercontent.com/45409515/161331763-f429b55d-c373-4e08-a458-3e8f2b5dbf12.png">


- However, I also tested all the other regex chars. 
<img width="1249" alt="Screen Shot 2022-04-01 at 3 52 37 PM" src="https://user-images.githubusercontent.com/45409515/161332565-7b19b8be-bc23-46ce-8c21-ca8482a34765.png">
